pkg/util/wait/wait_test.go

```diff
@@ -0,0 +1,55 @@
+package wait
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+)
+
+func TestForHTTPStatusOk(t *testing.T) {
+	tests := []struct {
+		name               string
+		expectedRespStatus []int
+		timeout            time.Duration
+		wantErr            bool
+	}{
+		{
+			name:               "working",
+			expectedRespStatus: []int{200},
+			timeout:            2 * time.Second,
+			wantErr:            false,
+		},
+		{
+			name:               "an error then ok",
+			expectedRespStatus: []int{500, 200},
+			timeout:            2 * time.Second,
+			wantErr:            false,
+		},
+		{
+			name:    "no response",
+			timeout: 1 * time.Second,
+			wantErr: true,
+		},
```

Comment:
I am wondering how can we structure these tests to allow unit-testing the syscall.ENETUNREACH and err.Timeout() || err.Err == io.EOF || err.Err == io.ErrUnexpectedEOF branches too. We probably need to mock `cli.Do(req)` which then would not require the current httptest setup that you have currently.